sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt

```diff
@@ -0,0 +1,57 @@
+package io.sentry.android.navigation
+
+import android.os.Bundle
+import androidx.navigation.NavController
+import androidx.navigation.NavDestination
+import io.sentry.Breadcrumb
+import io.sentry.Hint
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.SentryLevel.INFO
+import io.sentry.TypeCheckHint
+import java.lang.ref.WeakReference
+
+class SentryNavigationListener(
+    private val hub: IHub = HubAdapter.getInstance()
+) : NavController.OnDestinationChangedListener {
+
+    private var fromRef: WeakReference<NavDestination>? = null
+    private var fromArgs: Bundle? = null
+
+    override fun onDestinationChanged(
+        controller: NavController,
+        destination: NavDestination,
+        arguments: Bundle?
+    ) {
+        addBreadcrumb(destination, arguments)
+        fromRef = WeakReference(destination)
+        fromArgs = arguments
+    }
+
+    private fun addBreadcrumb(destination: NavDestination, arguments: Bundle?) {
+        val breadcrumb = Breadcrumb().apply {
+            type = "navigation"
+            category = "navigation"
+
+            val from = fromRef?.get()?.route
+            from?.let { data["from"] = it }
+            fromArgs?.let { args ->
+                data["from_arguments"] = args.keySet().filter {
```

Comment:
I think we should not send an empty map here, otherwise it doesn't look nice, wdyt?
![image](https://user-images.githubusercontent.com/4999776/174654551-ea58b827-bde8-4a0d-a9f0-19c72766a50f.png)